pages/home.js

```diff
@@ -3,101 +3,101 @@ import { Container } from "@mui/system";
-                return item;
-            })
-        );
-        setNfts(items);
-        setLoadingState("loaded");
-    }
-
-    //retrieve users saved books and map
+  // Load gloable user context
+  const { loaded, profile } = useContext(AppContext);
+  const [nfts, setNfts] = useState([]);
+  const [loadingState, setLoadingState] = useState("not-loaded");
+  useEffect(() => {
+    loadNFTs();
+  }, []);
+  async function loadNFTs() {
+    /* create a generic provider and query for unsold market items */
+    const provider = new ethers.providers.JsonRpcProvider();
+    const contract = new ethers.Contract(
+      marketplaceAddress,
+      BookMarketplace.abi,
+      provider
+    );
+    const data = await contract.fetchListings();
+    /*
+     *  map over items returned from smart contract and format
+     *  them as well as fetch their token metadata
+     */
+    const items = await Promise.all(
+      data.map(async (i) => {
+        const tokenUri = await contract.tokenURI(i.tokenId);
+        const meta = await axios.get(tokenUri);
+        const courseMeta = await Promise.all(
+          meta.data.courses.map(async (i) => {
+            const courseName = await axios.get(i);
 
-    const renderedListings = nfts.map((e, i) => {
-        return (
-            <SavedListing
-                key={i}
-                imgurl={e.image}
-                title={e.title}
-                author={e.author}
-                isbn={`ISBN: ${e.ISBN}`}
-                courses={e.course}
-                allowBid={e.allowBid}
-                pricing={e.price}
-                startingPrice={e.startingPrice}
-                retails="90"
-            />
+            return { name: courseName.data.name, URL: i };
+          })
         );
-    });
 
-    const renderListings = () => {
-        if (loadingState === "loaded" && !nfts.length)
-            return <Typography>Nothing to show yet</Typography>;
+        let item = {
+          price: ethers.utils.formatEther(i.instantPrice),
+          tokenId: i.tokenId.toNumber(),
+          seller: i.seller,
+          owner: i.owner,
+          allowBid: i.allowBid,
+          startingPrice: ethers.utils.formatEther(i.startingPrice),
+          image: meta.data.image,
+          title: meta.data.title,
+          author: meta.data.author,
+          ISBN: meta.data.ISBN,
+          course: courseMeta,
+        };
+        return item;
+      })
+    );
+    setNfts(items);
+    setLoadingState("loaded");
+  }
 
-        return <div className="book-shelf">{renderedListings}</div>;
-    };
+  //retrieve users saved books and map
 
+  const renderedListings = nfts.map((e, i) => {
     return (
-        <Container>
-            <ResponsiveAppBar />
-            <div className="titleHeader">
-                <h1 className="title">Your Saved Listings</h1>
-                {renderListings()}
-            </div>
-        </Container>
+      <div class="col-md-4">
+        <SavedListing
+          key={i}
```

Comment:
Hey I think you forgot to move the key props to the outside div. This way react can keep track of what to render next.